pandas/tests/formats/test_format.py

```diff
@@ -3925,6 +3925,10 @@ def test_rounding(self):
         result = formatter(0)
         self.assertEqual(result, u(' 0.000'))
 
+    def test_nan(self):
+        formatter = fmt.EngFormatter(accuracy=1, use_eng_prefix=True)
+        result = formatter(np.nan)
+        self.assertEqual(result, u('NaN'))
 
```

Comment:
Is this correct?

    def test_nan(self):
        formatter = fmt.EngFormatter(accuracy=1, use_eng_prefix=True)
        result = formatter(np.nan)
        self.assertEqual(result, u('NaN'))

        df = pd.DataFrame({'a':[1.5, 10.3, 20.5], 'b':[50.3, 60.67, 70.12], 'c':[100.2, 101.33, 120.33]})
        pt = df.pivot_table(values='a', index='b', columns='c')
        fmt.set_eng_float_format(accuracy=1)
        result = pt.to_string()
        print(result)
        
        expected = ('c         100.2E+00  101.3E+00  120.3E+00\n'
                    'b                                        \n'
                    '50.3E+00    1.5E+00        NaN        NaN\n'
                    '60.7E+00        NaN   10.3E+00        NaN\n'
                    '70.1E+00        NaN        NaN   20.5E+00')
        self.assertEqual(result, expected)